draft-ietf-tls-esni.md

```diff
@@ -842,6 +858,29 @@ example, if a client-facing server services many backend origin hosts, only one
 of which supports some cipher suite, it may be possible to identify that host
 based on the contents of unencrypted handshake messages.
 
+Beyond these primary security and privacy goals, ECH also aims to hide, to some
+extent, (a) whether or not a specific server supports ECH and (b) whether or
+not ECH was accepted for a particular connection. ECH aims to achieve both
+properties, assuming the attacker is passive and does not know the set of ECH
+configurations offered by the client-facing server. It does not achieve these
+properties for active attackers. More specifically:
+
+- Passive attackers with a known ECH configuration can distinguish between a
+connection that negotiates ECH with that configuration and one which does not,
+i.e., because the latter used a GREASE "encrypted_client_hello" extension (as
+specified in {{grease-extensions}}) or a different ECH configuration.
+- Passive attackers without the ECH configuration cannot distinguish between a
+connection that negotiates ECH and one which uses a GREASE
+"encrypted_client_hello" extension.
```

Comment:
Well, sort of. If they have connection memory they can see that everyone is using different config hashes.

Actually, I wonder if we should salt the hash.

Reply:
We can, yeah, but let's do that in a separate PR. (I'm assuming "without the ECH configuration" means it doesn't keep state.)

Reply:
OK